producer.go

```diff
@@ -797,9 +797,20 @@ func (p *producer) pollForSettingChanges(ctx context.Context, wg *sync.WaitGroup
 		case <-ctx.Done():
 			return
 		case <-ticker.C:
-			updatedQueue, err := p.fetchQueueSettings(ctx)
+			updatedQueue, err := func() (*rivertype.Queue, error) {
+				ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
+				defer cancel()
+
+				return p.exec.QueueGet(ctx, &riverdriver.QueueGetParams{
+					Name:   p.config.Queue,
+					Schema: p.config.Schema,
+				})
+			}()
```

Comment:
@bgentry Are you okay on inlining some of these trivial functions like this? IMO makes it easier to see what it's going to do without having to go search for the helper's implementation elsewhere in the file.